### PR TITLE
Css file restructure

### PR DESCRIPTION
**Pull request by kevinforrestconnors**
This merge adds:
- A [SMACSS](https://smacss.com/)-like SASS file structure 
- A button module
- Some basic styles to be used across the entire site, such as `box-sizing: border box`
- Adds `gem 'normalize-rails'` which [resets browser base styles ](https://github.com/markmcconachie/normalize-rails/blob/master/vendor/assets/stylesheets/normalize-rails/normalize.css)

This merge does not change any files that aren't in the `/app/assets/stylesheets` or `/app/views` directories, with the exception of the Gemfile

### CODE REVIEW

**Comment by alexthemitchell** [comment]
Please `bundle install` and commit the `Gemfile.lock` to resolve Travis CI build issues.

File: Gemfile [hunk 1]
```diff
@@ -20,6 +20,8 @@ gem 'twilio-ruby'
 gem 'rubocop'
 gem 'simplecov'
 
+gem 'normalize-rails'
```
Comment (alexthemitchell) [inline Gemfile:24]: Can you add a comment explaining what this gem does and why we use it?

File: app/assets/stylesheets/application.css.scss [hunk 2]
```diff
@@ -0,0 +1,4 @@
+//= require normalize-rails
```
Comment (alexthemitchell) [inline app/assets/stylesheets/application.css.scss:1]: Are these comments? If so, are they necessary?

File: app/assets/stylesheets/layout/login.css.scss [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/* header */
```
Comment (alexthemitchell) [inline app/assets/stylesheets/layout/login.css.scss:1]: Is this just boilerplate?

File: app/assets/stylesheets/application.css.scss (hunk 2 above)
Comment (kevinforrestconnors) [inline app/assets/stylesheets/application.css.scss:1]: No, these are not comments and they are necessary.
The compass gem http://compass-style.org/ parses lines in `application.css.scss` for anything starting with `//=` and then includes them in the build.

File: app/assets/stylesheets/layout/login.css.scss (hunk 3 above)
Comment (kevinforrestconnors) [inline app/assets/stylesheets/layout/login.css.scss:1]: Yes.

File: Gemfile [hunk 4]
```diff
@@ -20,6 +20,8 @@ gem 'twilio-ruby'
 gem 'rubocop'
 gem 'simplecov'
 
+gem 'normalize-rails' # resets base browser styles. see this url for the styles added. https://goo.gl/SzZnY2
```
Comment (alexthemitchell) [inline Gemfile]: Can we put the comment on the line above? Also, this comment is not helpful at all. I have absolutely no idea what a "base browser style" is, nor why I would reset it. If nothing else, at least tag that it's related to front-end stylesheets.

File: app/assets/stylesheets/layout/login.css.scss (hunk 3 above)
Comment (alexthemitchell) [inline app/assets/stylesheets/layout/login.css.scss:1]: Remove.

**Comment by alexthemitchell** [comment]
@kevinforrestconnors If you can just clarify that comment (and put it on the previous line) then this LGTM. Provided Travis builds after that next commit, this should be merged. You will need to deal with merge conflicts.

**Comment by alexthemitchell** [comment]
LGTM. Fix merge conflicts and then you're good to go.

**Pull request merged**